evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -98,19 +94,18 @@ def transform(self, X, y=None):
             ww.DataTable: Transformed X
         """
         X_ww = infer_feature_types(X)
-        X_null_dropped = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
-        X_null_dropped.drop(self._all_null_cols, inplace=True, axis=1, errors='ignore')
+        X_null_dropped = X_ww.ww.drop(self._all_null_cols)
         if X_null_dropped.empty:
-            return _retain_custom_types_and_initalize_woodwork(X_ww, X_null_dropped)
+            return X_null_dropped
 
         if self._numeric_cols is not None and len(self._numeric_cols) > 0:
-            X_numeric = X_null_dropped[self._numeric_cols]
-            imputed = self._numeric_imputer.transform(X_numeric).to_dataframe()
+            X_numeric = X_null_dropped.ww[self._numeric_cols.tolist()]
+            imputed = self._numeric_imputer.transform(X_numeric)
             X_null_dropped[X_numeric.columns] = imputed
 
         if self._categorical_cols is not None and len(self._categorical_cols) > 0:
-            X_categorical = X_null_dropped[self._categorical_cols]
-            imputed = self._categorical_imputer.transform(X_categorical).to_dataframe()
+            X_categorical = X_null_dropped.ww[self._categorical_cols.tolist()]
+            imputed = self._categorical_imputer.transform(X_categorical)
             X_null_dropped[X_categorical.columns] = imputed
-        X_null_dropped = _retain_custom_types_and_initalize_woodwork(X_ww, X_null_dropped)
+        X_null_dropped = _retain_custom_types_and_initalize_woodwork(X_ww.ww.logical_types, X_null_dropped)
```

Comment:
I just read [the impl of `_retain_custom_types_and_initialize_woodwork` on this feature branch](https://github.com/alteryx/evalml/blob/2035-use-ww-accessor/evalml/utils/woodwork_utils.py#L79). And yeah, not sure I follow just yet. I'd appreciate it if you could explain to me how the pieces fit together. We can do that here or offline.

Reply:
The impl of `_retain_custom_types_and_initialize_woodwork` on main does not preserve semantic tags either.

I agree it'd be good to do that but I think it'll be complicated:

1. Not sure if there's a way to tell woodwork to redo inference for a particular column and carry over the semantic tags. It's probably doable on our end but it would complicate the impl of `_retain_custom_types_`
2. Our `ComponentGraph` uses `concat` [under the hood](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/component_graph.py#L176) which will not preserve the schema even if we do so in every component. I think that's blocked on a way to concat through woodwork.

Reply:
Hmm ok. Let's catch up on this soon.